pkgs/development/python-modules/pypass/default.nix

```diff
@@ -0,0 +1,89 @@
+{ buildPythonPackage
+, click
+, colorama
+, enum34
+, fetchFromGitHub
+, fetchgit
+, fetchPypi
+, git
+, gnugrep
+, gnupg
+, nose
+, pbr
+, pexpect
+, pythonAtLeast
+, pythonOlder
+, stdenv
+, tree
+, xclip
+}:
+
+# NOTE: pypass can also be used as an application, but probably the most
+# important usecase is as a library. So, let's use buildPythonPackage and
+# support any Python version instead of defining it as an application with
+# buildPythonApplication.
+buildPythonPackage rec {
+  pname = "pypass";
+  version = "0.2.1";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "1nm4mj7pd7gz4ghic6b3wrnd1b59hd1f0axavdabfl79wy511l7r";
+  };
+
+  patches = [ ./mark-executables.patch ];
+
+  # Paths to executables that pypass uses
+  git_exec = "${git}/bin/git";
+  grep_exec = "${gnugrep}/bin/grep";
+  gpg_exec = "${gnupg}/bin/gpg2";
+  tree_exec = "${tree}/bin/tree";
+  xclip_exec = "${xclip}/bin/xclip";
+
+  # Remove enum34 requirement if Python >= 3.4. Fix the paths to some
+  # executables that the Python code runs as subprocesses.
+  postPatch = ''
+    substituteAllInPlace Makefile
+    substituteAllInPlace pypass/command.py
+    substituteAllInPlace pypass/passwordstore.py
+    substituteAllInPlace pypass/tests/test_command.py
+    substituteAllInPlace pypass/tests/test_passwordstore.py
```

Comment:
This is all an anti pattern. Import `substituteAll` in the arguments at the top and use:

```nix
  patches = [
    (substitueAll {
      src = ./mark-executables.patch;
      git_exec = "${git}/bin/git";
      grep_exec = "${gnugrep}/bin/grep";
      gpg_exec = "${gnupg}/bin/gpg2";
      tree_exec = "${tree}/bin/tree";
      xclip_exec = "${xclip}/bin/xclip";
    })
  ];
```

Reply:
Oh, wow, thanks! :+1: I wasn't aware of this! :gem:

Reply:
Fixed.